lib/Sema/CSDiagnostics.cpp

```diff
@@ -1328,6 +1328,37 @@ bool MissingExplicitConversionFailure::diagnoseAsError() {
   return true;
 }
 
+/// Returns true if \p locator is ending with either of the following
+///  - Member
+///  - Member -> KeyPathDynamicMember
+static bool isLocatorPointingToMember(ConstraintLocator *locator) {
```

Comment:
Nit: I think this is useful to have on the locator itself as `->isMemberRef()` or something like that.

Reply:
Good idea. Changed it 👍🏽